### PR TITLE
dns_utils: allow an optional DNS server IP in DNS_PUBLIC

### PR DESCRIPTION
**Pull request by moneromooo-monero**
tcp://a.b.c.d

### CODE REVIEW

File: src/common/dns_utils.cpp [hunk 1]
```diff
@@ -207,14 +209,31 @@ DNSResolver::DNSResolver() : m_data(new DNSResolverData())
       LOG_PRINT_L0("Using public DNS server: " << dns_public_addr << " (TCP)");
       use_dns_public = 1;
     }
+    else if (sscanf(dns_public.c_str(), "tcp://%u.%u.%u.%u%c", &ip0, &ip1, &ip2, &ip3, &c) == 4)
```
Comment (MaxXor) [inline src/common/dns_utils.cpp]: Why this additional `char c`? I tested it and it works also without it.

Comment (moneromooo-monero) [reply src/common/dns_utils.cpp]: Try tcp://1.2.3.4x with and without.

Comment (MaxXor) [reply src/common/dns_utils.cpp]: Ah I see it detects it as invalid. :)

Comment (erikd) [reply src/common/dns_utils.cpp]: Is it worth adding a test case so that if someone removes the `%c` the test fails?

Comment (moneromooo-monero) [reply src/common/dns_utils.cpp]: done

**Pull request merged**